File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/UserCouponRepository.java
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository;

import com.eager.questioncloud.core.domain.coupon.dto.AvailableUserCoupon;
import com.eager.questioncloud.core.domain.coupon.model.UserCoupon;
import java.util.List;

public interface UserCouponRepository {
    UserCoupon getUserCoupon(Long userCouponId, Long userId);

    UserCoupon getUserCoupon(Long userCouponId);

    Boolean isRegistered(Long userId, Long couponId);

    UserCoupon save(UserCoupon userCoupon);

    List<AvailableUserCoupon> getAvailableUserCoupons(Long userId);

    Boolean use(Long userCouponId);

    void deleteAllInBatch();
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/UserCouponRepository.kt
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository

import com.eager.questioncloud.core.domain.coupon.dto.AvailableUserCoupon
import com.eager.questioncloud.core.domain.coupon.model.UserCoupon

interface UserCouponRepository {
    fun getUserCoupon(userCouponId: Long, userId: Long): UserCoupon

    fun getUserCoupon(userCouponId: Long): UserCoupon

    fun isRegistered(userId: Long, couponId: Long): Boolean

    fun save(userCoupon: UserCoupon): UserCoupon

    fun getAvailableUserCoupons(userId: Long): List<AvailableUserCoupon>

    fun use(userCouponId: Long): Boolean

    fun deleteAllInBatch()
}
